src/Workspaces/Core/Portable/Workspace/Host/TaskScheduler/WorkspaceTaskSchedulerFactory.WorkspaceTaskScheduler.cs

```diff
@@ -36,29 +36,25 @@ public WorkspaceTaskScheduler(WorkspaceTaskSchedulerFactory factory, TaskSchedul
             public Task ScheduleTask(Action taskAction, string taskName, CancellationToken cancellationToken)
             {
                 return ScheduleTaskWorker(
-                    taskName, () => Task.Factory.SafeStartNew(taskAction, cancellationToken, _taskScheduler),
-                    cancellationToken);
+                    taskName, () => Task.Factory.SafeStartNew(taskAction, cancellationToken, _taskScheduler));
```

Comment:
i feel dumb... why is this better than before?  the lack of cancellationToken seems fishy at first look...

Reply:
The CancellationToken being passed wasn't being used. :smile: And correctly so: in this case ScheduleTaskWorker is nothing more than a wrapper around the process of adding an async token for tracking the work; there's nothing that can consume cancellation there.

Reply:
It might make more sense if you compare to [the equivalent Queue implementation](https://github.com/dotnet/roslyn/blob/9e04d346b87e8eb7f8eda5852bf5d107e2e664a7/src/Workspaces/Core/Portable/Workspace/Host/TaskScheduler/WorkspaceTaskSchedulerFactory.WorkspaceTaskQueue.cs)